packages/playwright-core/src/utils/isomorphic/selectorParser.ts

```diff
@@ -123,11 +123,18 @@ function selectorPartsEqual(list1: ParsedSelectorPart[], list2: ParsedSelectorPa
   return stringifySelector({ parts: list1 }) === stringifySelector({ parts: list2 });
 }
 
-export function stringifySelector(selector: string | ParsedSelector): string {
+export function stringifySelector(selector: string | ParsedSelector, forceEngineName?: boolean): string {
   if (typeof selector === 'string')
     return selector;
   return selector.parts.map((p, i) => {
-    const prefix = p.name === 'css' ? '' : p.name + '=';
+    let hideEngine = false;
+    if (!forceEngineName) {
```

Comment:
While reading this, I realized that we cannot hide engine for the "capture" part.

```suggestion
    if (!forceEngineName && i !== selector.capture) {
```